### ISSUE TITLE
Build Error, debugpy version debugpy-1.2.1.zip from PyPi

### ISSUE THREAD

**Issue by RobertSwirsky** [microsoft/debugpy#555]
## Environment data

- debugpy version: debugpy-1.2.1.zip 
- OS and version: Build OS: 5.8.0-44-generic #50~20.04.1-Ubuntu SMP Wed Feb 10 21:07:30 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux
- Python version (& distribution if applicable, e.g. Anaconda): Python 3.8.5 (default, Jan 27 2021, 15:41:15)
- Using VS Code or Visual Studio: No 

## Actual behavior

		Cannot compile usr/lib/python3.8/site-packages/debugpy/_vendored/pydevd/_pydev_imps/_pydev_SimpleXMLRPCServer.py
		Traceback (most recent call last):
		  File "/home/robert/Repos/ptpmaster/buildroot/output/host/lib/python3.8/py_compile.py", line 144, in compile
		    code = loader.source_to_code(source_bytes, dfile or file,
		  File "<frozen importlib._bootstrap_external>", line 812, in source_to_code
		  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
		  File "usr/lib/python3.8/site-packages/debugpy/_vendored/pydevd/_pydev_imps/_pydev_SimpleXMLRPCServer.py", line 263
		    except Fault, fault:
		                ^
    SyntaxError: invalid syntax

XXX

## Expected  behavior

(Code should compile)

## Steps to reproduce:

    1. Compile latest source tar from PyPi

**Comment by int19h** [comment]
As the comment in that file explains, it's a copy of a stdlib module that's intended to be used on Jython, because it's missing that part of stdlib. As such, it's written in Python 2, and cannot be run using Python 3. It is conditionally imported only if the standard module is not available, which should not be the case on any version of Python supported by debugpy.

Also, this is not something that happens if you use `setup.py bdist` or `bdist_wheel`, which is the only build that we officially maintain and support. I'm not sure what build system you're using, and why it's trying to compile or otherwise randomly syntax-check Python files, but that approach is not going to work - even if you get past this, there are other parts of the debugger that are written in Python 2 syntax and are similarly conditionally loaded, because that was the only way to make it work across both major versions (precisely because of such syntactic incompatibilities making it impossible to do simple inline version checks).

Basically, the build system should not blindly assume that every .py file in the package is valid code for whatever Python version it's trying to build the package for, and try to parse it as such. Or, at least, it shouldn't be treating it as a fatal error if that happens. As evidenced by the fact that the package actually *runs*, it's perfectly valid Python code.

**Comment by RobertSwirsky** [comment]
Thank you for your answer, int19h!

I'm trying to build this using "buildroot" as a cross-compile. I guess I don't know buildroot well enough. I'll see what I can do to make buildroot give the project the correct settings.
